File: src/main/java/com/ballgame/WallBarrier.java
package com.ballgame;

import java.awt.Graphics2D;
import java.awt.Shape;

public class WallBarrier extends Rectangle {
    static enum Side {
        LEFT,
        RIGHT
    }

    private Shape shape;
    private double velX;
    private double velY;
    private Side side;

    public WallBarrier(int width, int height, Side side) {
        super(0, 0);

        if (side == Side.LEFT)
            this.x = 150;
        else if (side == Side.RIGHT)
            this.x = Eskiv.WIDTH - width - 150;
        this.side = side;
        this.width = width;
        this.height = height;
        this.velY = 0;
        this.velX = 0;

        this.shape = new java.awt.Rectangle.Double((int) this.x, (int) this.y, this.width, this.height);
    }

    public Side getSide() {
        return this.side;
    }

    public Shape getShape() {
        return this.shape;
    }

    public double getVelX() {
        return this.velX;
    }

    public double getVelY() {
        return this.velY;
    }

    public void setVelX(double velX) {
        this.velX = velX;
    }

    public void setVelY(double velY) {
        this.velY = velY;
    }

    public void update(double deltaTime, double previousTime) {
        double incrementX = this.velX * (deltaTime - previousTime);
        double incrementY = this.velY * (deltaTime - previousTime);

        if ((this.side == Side.LEFT || this.side == Side.RIGHT) && this.y + this.height >= Eskiv.HEIGHT
                && this.velY > 0) {
            this.velY *= -1;
        } else if ((this.side == Side.LEFT || this.side == Side.RIGHT) && this.y <= 0 && this.velY < 0) {
            this.velY *= -1;
        }

        super.setTransform(this.x + incrementX, this.y + incrementY);
    }

    public void render(Graphics2D g2d) {
        java.awt.Rectangle.Double obj = (java.awt.Rectangle.Double) this.shape;
        obj.setFrame(this.x, this.y, this.width, this.height);
        g2d.fill(this.shape);
    }
}

File: src/main/kotlin/com/ballgame/WallBarrier.kt
package com.ballgame

import com.ballgame.Eskiv.Companion.HEIGHT
import com.ballgame.Eskiv.Companion.WIDTH
import com.ballgame.WallBarrier.Companion.Side.LEFT
import com.ballgame.WallBarrier.Companion.Side.RIGHT
import java.awt.Graphics2D
import java.awt.Shape
import java.awt.geom.Rectangle2D
import java.awt.geom.Rectangle2D.*
import kotlin.Double

class WallBarrier(width: Int, height: Int, side: Side) : Rectangle(0, 0) {
	val shape: Shape
	private var velX: Double
	var velY: Double
	val side: Side

	init {
		if (side == LEFT) x = 150.0 else if (side == RIGHT) x = (WIDTH - width - 150).toDouble()
		this.side = side
		this.width = width
		this.height = height
		velY = 0.0
		velX = 0.0
		shape = Double(x, y, this.width.toDouble(), this.height.toDouble())
	}

	override fun update(deltaTime: Double, previousTime: Double) {
		if (((side == LEFT) || (side == RIGHT)) && ((y + height) >= HEIGHT) && (velY > 0)) velY *= -1.0 else if ((side == LEFT || side == RIGHT) && y <= 0 && velY < 0) velY *= -1.0
		super.setTransform(x + (velX * (deltaTime - previousTime)), y + (velY * (deltaTime - previousTime)))
	}

	override fun render(g2d: Graphics2D?) {
		val obj = shape as Rectangle2D.Double
		obj.setFrame(x, y, width.toDouble(), height.toDouble())
		g2d!!.fill(shape)
	}

	companion object {
		enum class Side {
			LEFT, RIGHT
		}
	}
}
